cvat/apps/engine/serializers.py

```diff
@@ -1207,6 +1206,72 @@ class SubLabeledShapeSerializer(ShapeSerializer, AnnotationSerializer):
 class LabeledShapeSerializer(SubLabeledShapeSerializer):
     elements = SubLabeledShapeSerializer(many=True, required=False)
 
+class LabeledImageSerializerFromDB(serializers.BaseSerializer):
+    def to_representation(self, instance):
+        def convert_tag(tag):
+            attr_keys = ['spec_id', 'value']
+            keys = ['id', 'label_id', 'frame', 'group', 'source']
+
+            result = OrderedDict([(key, tag[key]) for key in keys])
+            result['attributes'] = [
+                OrderedDict([(key, attr[key]) for key in attr_keys]) for attr in tag['labeledimageattributeval_set']
+            ]
+
+            return result
+
+        return convert_tag(instance)
+
+class LabeledShapeSerializerFromDB(serializers.BaseSerializer):
+    def to_representation(self, instance):
+        def convert_shape(shape):
+            attr_keys = ['spec_id', 'value']
+            keys = [
+                'id', 'label_id', 'type', 'frame', 'group',
+                'source', 'occluded', 'outside', 'z_order',
+                'rotation', 'points',
+            ]
+            result = OrderedDict([(key, shape[key]) for key in keys])
+            result['attributes'] = [
+                OrderedDict([(key, attr[key]) for key in attr_keys]) for attr in shape['labeledshapeattributeval_set']
+            ]
```

Comment:
There's a lot of duplication between the three classes. I think some common functions could be extracted here so that this code would become something like this:

```python
result = _convert_annotation(shape, specific_keys=['type', 'occluded', 'outside', 'z_order', 'rotation', 'points'])
result['attributes'] = _convert_attributes(shape['labeledshapeattributeval_set'])
```